pkg/katapult/virtual_machine_group.go

```diff
@@ -14,13 +15,53 @@ type VirtualMachineGroup struct {
 	CreatedAt *timestamp.Timestamp `json:"created_at,omitempty"`
 }
 
-type VirtualMachineGroupsClient struct {
-	client   *apiClient
-	basePath *url.URL
+func (s *VirtualMachineGroup) lookupReference() *VirtualMachineGroup {
+	if s == nil {
+		return nil
+	}
+
+	return &VirtualMachineGroup{ID: s.ID}
+}
+
+func (s *VirtualMachineGroup) queryValues() *url.Values {
+	v := &url.Values{}
+
+	if s != nil {
+		v.Set("virtual_machine_group[id]", s.ID)
+	}
+
+	return v
+}
+
+type VirtualMachineGroupCreateArguments struct {
+	Name      string
+	Segregate *bool
+}
+
+type VirtualMachineGroupUpdateArguments struct {
+	Name      string
+	Segregate *bool
+}
+
+type virtualMachineGroupCreateRequest struct {
+	Organization *Organization `json:"organization,omitempty"`
+	Name         string        `json:"name,omitempty"`
+	Segregate    *bool         `json:"segregate,omitempty"`
+}
+
+type virtualMachineGroupUpdateRequest struct {
+	VirtualMachineGroup *VirtualMachineGroup `json:"virtual_machine_group,omitempty"`
+	Name                string               `json:"name,omitempty"`
+	Segregate           *bool                `json:"segregate,omitempty"`
 }
 
 type virtualMachineGroupsResponseBody struct {
 	VirtualMachineGroups []*VirtualMachineGroup `json:"virtual_machine_groups,omitempty"`
+	VirtualMachineGroup  *VirtualMachineGroup   `json:"virtual_machine_group,omitempty"`
+}
+type VirtualMachineGroupsClient struct {
```

Comment:
Interesting, I would have though the lack of a blank line here would have been caught by linting. I'll take a look at the linting config, but in the meantime:

```suggestion
}

type VirtualMachineGroupsClient struct {
```